src/Inciter/AMR/node_store.h

```diff
@@ -0,0 +1,245 @@
+#ifndef AMR_node_store_h
+#define AMR_node_store_h
+
+#include <cmath>
+
+#include "types.h"
+#include "tet_store.h"
+
+// TODO: make this have a base class to support multiple generator schemes
+// using the policy design pattern
+namespace AMR {
+
+    class node_store_t
```

Comment:
I wonder how this might be possible to merge to the existing [UnsMesh](https://github.com/quinoacomputing/quinoa/blob/master/src/Mesh/UnsMesh.h) class.